### ISSUE TITLE
crash in GetAccountInformationAsync under mono

### ISSUE THREAD

**Issue by Ben8823** [mfilippov/vimeo-dot-net#115]
I'm using your library for video upload on vimeo. Under .net everything works fine but under ubuntu 16.04 with mono (version 5.4) I get the following exception when calling vimeoClient.GetAccountInformationAsync(). Any idea what could be wrong?

```
 One or more errors occurred. ---> System.ArgumentNullException: Value cannot be null.
                           Parameter name: exception
                             at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[TResult].SetException (System.Exception exception) [0x00003] in <62f5937022004555807e6c57c33f6684>:0 
                             at VimeoDotNet.VimeoClient+<GetAccountInformationAsync>d__63.MoveNext () [0x0009d] in <29be34c8defa4f729d3a008aa46c551e>:0 
                             at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[TResult].Start[TStateMachine] (TStateMachine& stateMachine) [0x0002c] in <62f5937022004555807e6c57c33f6684>:0 
                             at VimeoDotNet.VimeoClient.GetAccountInformationAsync () [0x00023] in <29be34c8defa4f729d3a008aa46c551e>:0 
                             at wn_remoted.RemoteServer+<HandleVimeoUpload>d__47.MoveNext () [0x00036] in <5ed6f39be0a24180880c2e1569edb454>:0 
                              --- End of inner exception stack trace ---
                             at System.Threading.Tasks.Task.ThrowIfExceptional (System.Boolean includeTaskCanceledExceptions) [0x00011] in <62f5937022004555807e6c57c33f6684>:0 
                             at System.Threading.Tasks.Task.Wait (System.Int32 millisecondsTimeout, System.Threading.CancellationToken cancellationToken) [0x00043] in <62f5937022004555807e6c57c33f6684>:0 
                             at System.Threading.Tasks.Task.Wait () [0x00000] in <62f5937022004555807e6c57c33f6684>:0 
                             at wn_remoted.RemoteServer.HandleUploadFile (Newtonsoft.Json.Linq.JObject data) [0x00094] in <5ed6f39be0a24180880c2e1569edb454>:0 
                             at wn_remoted.RemoteServer.ws_OnMessage (System.Object sender, WebSocketSharp.MessageEventArgs e) [0x003c2] in <5ed6f39be0a24180880c2e1569edb454>:0 
                             at WebSocketSharp.Ext.Emit[TEventArgs] (System.EventHandler`1[TEventArgs] eventHandler, System.Object sender, TEventArgs e) [0x00009] in <38d3cef14c5a4fc9a92de0991034bc1a>:0 
                             at WebSocketSharp.WebSocket.messagec (WebSocketSharp.MessageEventArgs e) [0x00003] in <38d3cef14c5a4fc9a92de0991034bc1a>:0 
                           ---> (Inner Exception #0) System.ArgumentNullException: Value cannot be null.
                           Parameter name: exception
                             at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[TResult].SetException (System.Exception exception) [0x00003] in <62f5937022004555807e6c57c33f6684>:0 
                             at VimeoDotNet.VimeoClient+<GetAccountInformationAsync>d__63.MoveNext () [0x0009d] in <29be34c8defa4f729d3a008aa46c551e>:0 
                             at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[TResult].Start[TStateMachine] (TStateMachine& stateMachine) [0x0002c] in <62f5937022004555807e6c57c33f6684>:0 
                             at VimeoDotNet.VimeoClient.GetAccountInformationAsync () [0x00023] in <29be34c8defa4f729d3a008aa46c551e>:0 
                             at wn_remoted.RemoteServer+<HandleVimeoUpload>d__47.MoveNext () [0x00036] in <5ed6f39be0a24180880c2e1569edb454>:0 <---
```

**Comment by mfilippov** [comment]
@Ben8823  Thanks for the feedback. I try to reproduce the problem in the next weekend.

**Comment by Ben8823** [comment]
Hi, I've found the issue in the meantime.

The setter for Path in ApiRequest.cs calls Uri.TryCreate(value, UriKind.Absolute, out parsed) with value = "/me" which correctly returns false in .net but true in mono, which causes a infinite recursion. It's probably a mono bug but I've added a second check as a workaround for mono: !value.StartsWith("/") 

With that change everything works fine under mono. Maybe you can include a fix in one of your next releases?

**Comment by mfilippov** [comment]
@Ben8823 Thank you for the information. Issue fixed in 1.0.0.8.
